src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -332,19 +315,35 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
                     break;
             }
 
+            if (isIntegratedSecurity)
+            {
+                string hostName = details.ServerName;
```

Comment:
Did you mean `GetSqlServerSPN(DataSource details)`?
It creates additional unnecessary dependency between `GetSqlServerSPN()` method `DataSource` object. `GetSqlServerSPN()` needs nothing but hostname and postfix string, and it does not need entire `DataSource` object that contains all the necessary information. The point I am making is some change on  `DataSource` object might lead to change on `GetSqlServerSPN()` due to coupling led by `GetSqlServerSPN(DataSource details)`